src/main/java/de/fraunhofer/aisec/cpg/TranslationConfiguration.java

```diff
@@ -79,6 +79,9 @@
 
   @NonNull private List<Pass> passes;
 
+  /** For test purposes, the cleanup phase after the analysis may want to be skipped */
+  private boolean cleanupOnCompletion;
```

Comment:
I am not a big fan of including functionality that is only needed during testing, especially in a very public facing class such as the configuration. Is there any other way to do this?

Reply:
In order to correctly test the common type finder, I need the TypeManager to hold all state that it collected when the graph has been built. Thus cleaning everything up is counterproductive here. Anyways I don't think that skipping cleanup has any bad consequences either.

Reply:
I'll do that differently and mock the cleanup method instead of creating a global config for it

Reply:
Yeah that sounds like a better solution.